shotover-proxy/src/message/mod.rs

```diff
@@ -20,325 +20,265 @@ use itertools::Itertools;
+    // Used for ordering out of order messages without parsing their contents.
+    // TODO: We will have a better idea of how to make this generic once we have multiple out of order protocols
+    //       For now its just written to match cassandra's stream_id field
+    pub fn stream_id(&self) -> Option<i16> {
+        match &self.inner {
+            Some(MessageInner::RawBytes {
+                bytes,
+                message_type: MessageType::Cassandra,
+            }) => {
+                const HEADER_LEN: usize = 9;
+                if bytes.len() >= HEADER_LEN {
+                    Some((&bytes[2..4]).get_i16())
                 } else {
-                    "UNKNOWN".to_string()
+                    None
                 }
             }
+            Some(MessageInner::RawBytes { .. }) => None,
+            Some(MessageInner::Parsed { frame, .. } | MessageInner::Modified { frame }) => {
+                match frame {
+                    Frame::Cassandra(cassandra) => Some(cassandra.stream_id),
+                    Frame::Redis(_) => None,
+                    Frame::None => None,
+                }
+            }
+            None => None,
         }
     }
 }
 
-#[derive(PartialEq, Debug, Clone)]
-pub struct QueryMessage {
-    pub query_string: String,
-    pub namespace: Vec<String>,
-    pub primary_key: HashMap<String, MessageValue>,
-    pub query_values: Option<HashMap<String, MessageValue>>,
-    pub projection: Option<Vec<String>>,
-    pub query_type: QueryType,
-    pub ast: Option<ASTHolder>,
-}
-
-impl QueryMessage {
-    pub fn empty() -> Self {
-        QueryMessage {
-            query_string: "".to_string(),
-            namespace: vec![],
-            primary_key: Default::default(),
-            query_values: None,
-            projection: None,
-            query_type: QueryType::Read,
-            ast: None,
-        }
-    }
-
-    pub fn get_namespace(&self) -> Vec<String> {
-        self.namespace.clone()
-    }
-
-    pub fn set_namespace_elem(&mut self, index: usize, elem: String) -> String {
-        let old = self.namespace.remove(index);
-        self.namespace.insert(index, elem);
-        old
-    }
-
-    pub fn get_primary_key(&self) -> Option<String> {
-        let f: Vec<String> = self
-            .primary_key
-            .iter()
-            .map(|(_, v)| serde_json::to_string(&v).unwrap())
-            .collect();
-        Some(f.join("."))
-    }
-
-    pub fn get_namespaced_primary_key(&self) -> Option<String> {
-        self.get_primary_key().map(|pk| {
-            let mut buffer = String::new();
-            let f = self.namespace.join(".");
-            buffer.push_str(f.as_str());
-            buffer.push('.');
-            buffer.push_str(serde_json::to_string(&pk).unwrap().as_str());
-            buffer
-        })
-    }
-}
+// TODO: When im finished, evaluate if MessageInner as an enum brings value over just a struct of Options
```

Comment:
did we do this ? Doesn't seem worthwhile compared to the semantic value the current enum brings

Reply:
Having a struct of Options would allow us to make `inner: Option<MessageInner>` into `inner: MessageInner`.

This could almost be an improvement however because of the MessageType needed in RawBytes representing MessageInner as a struct of Options would be quite problematic.

In conclusion both options are kinda bleh, thats the real world for you I guess.

I've deleted the comment, lets just leave MessageInner as it is.